---
# title: "Regulatory similarity analysis of 39 disease- and trait-associated SNP sets"
# author: "Mikhail Dozmorov"
# date: "`r format(Sys.time(), '%d %B, %Y')`"
output: html_document
---
```{r setup, echo=FALSE, include=FALSE, cache=FALSE, warning=FALSE, message=FALSE, eval=TRUE}
# Set up the environment
library(knitr) 
opts_chunk$set(cache.path='cache/', fig.path='img/', cache=F, tidy=T, fig.keep='high', dpi=100,  echo=F, autodep=T, width=200, warning=FALSE, comment=NA, results='hide', eval=FALSE) # Need to explicitly set eval=TRUE for chunks
options(replace.assign=TRUE, width=500)
tidy.opts=list(width.cutoff=200)
suppressMessages(library(pander))
panderOptions('table.split.table', Inf)
set.seed(1)
# Local functions and settings
library(MDmisc)
#source("/Users/mikhail/Documents/Work/GenomeRunner/R.GenomeRunner/utils2.R") # See the required packages there
#source("/Users/mikhail/Documents/Work/GenomeRunner/R.GenomeRunner/episimilarity.R")
# External libraries
library(dplyr)
# Correlation to use
cortype <- "spearman"
# Clustering: "ward.D", "ward.D2", "single", "complete", "average" (= UPGMA), "mcquitty" (= WPGMA), "median" (= WPGMC) or "centroid" (= UPGMC)
hclust.method <- "ward.D2"
# Color scheme
col3 <- colorRampPalette(c("blue", "yellow", "red"))
cells_blood_all <- c("E033", "E034", "E037", "E038", "E039", "E040", "E041", "E042", "E043", "E044", "E045", "E047", "E048", "E062", "E115", "E116", "E123", "E124", "E029", "E030", "E031", "E032", "E035", "E036", "E046", "E050", "E051")
```

### Load the data

```{r eval=TRUE}
library(xlsx)
# Load term mapping
term.mapping <- read.xlsx2("data/icd9_mapping.xlsx", sheetName="manual")
# term.mapping <- term.mapping[ term.mapping$BED != "Migraine", ] # Exclude  outlier
# Load the actual episimilarity data
fname <- c("data.gr.05/grweb_roadmapHistone_gPk-imputed/matrix_PVAL.txt", "data.gr.05/grweb_roadmapDNase_gPk-imputed/matrix_PVAL.txt") # Best correlation results
# fname <- c("data.gr.05/grweb_roadmapDNase_gPk-imputed//matrix_PVAL.txt", "data.gr.05/grweb_roadmapDNase_nPk-imputed//matrix_PVAL.txt", "data.gr.05/grweb_roadmapHistone_gPk-imputed//matrix_PVAL.txt", "data.gr.05/grweb_roadmapHistone_nPk-imputed//matrix_PVAL.txt") # Imputed-ALL
# fname <- c("data.gr.05/grweb_roadmapDNase_bPk-processed//matrix_PVAL.txt", "data.gr.05/grweb_roadmapDNase_nPk-processed//matrix_PVAL.txt", "data.gr.05/grweb_roadmapHistone_bPk-processed//matrix_PVAL.txt", "data.gr.05/grweb_roadmapHistone_gPk-processed//matrix_PVAL.txt", "data.gr.05/grweb_roadmapHistone_nPk-processed//matrix_PVAL.txt") # Processed-ALL
# fname <- c("data.gr.05/grweb_encBroadHmm//matrix_PVAL.txt", "data.gr.05/grweb_encChromCombined//matrix_PVAL.txt", "data.gr.05/grweb_encChromHmm//matrix_PVAL.txt", "data.gr.05/grweb_encDNase//matrix_PVAL.txt", "data.gr.05/grweb_encHistone//matrix_PVAL.txt", "data.gr.05/grweb_encSegway//matrix_PVAL.txt", "data.gr.05/grweb_encTFBS_cellspecific//matrix_PVAL.txt", "data.gr.05/grweb_encTFBS_clustered//matrix_PVAL.txt", "data.gr.05/grweb_roadmapChrom15//matrix_PVAL.txt", "data.gr.05/grweb_roadmapChrom18//matrix_PVAL.txt", "data.gr.05/grweb_roadmapChrom25//matrix_PVAL.txt", "data.gr.05/grweb_roadmapDNase_bPk-processed//matrix_PVAL.txt", "data.gr.05/grweb_roadmapDNase_gPk-imputed//matrix_PVAL.txt", "data.gr.05/grweb_roadmapDNase_nPk-imputed//matrix_PVAL.txt", "data.gr.05/grweb_roadmapDNase_nPk-processed//matrix_PVAL.txt", "data.gr.05/grweb_roadmapHistone_bPk-processed//matrix_PVAL.txt", "data.gr.05/grweb_roadmapHistone_gPk-imputed//matrix_PVAL.txt", "data.gr.05/grweb_roadmapHistone_gPk-processed//matrix_PVAL.txt", "data.gr.05/grweb_roadmapHistone_nPk-imputed//matrix_PVAL.txt", "data.gr.05/grweb_roadmapHistone_nPk-processed//matrix_PVAL.txt") # Complete-ALL
mtx <- gr_load_data(fname, p2z = T)
mtx <- mtx[, match(term.mapping$BED, colnames(mtx))] # Match and subset the data
# A vector of category names, same order as the correlation matrix
categoryNames <- term.mapping$Category
names(categoryNames) <- term.mapping$Name
# Trim column names
colnames(mtx) <- term.mapping$Name
```

```{r}
# We first check how the distributions of the values for each sample look like side-by-side.
ggplot(reshape2::melt(mtx), aes(x=Var2, y=value, fill=Var2)) + 
  geom_boxplot() + 
  theme(axis.text.x=element_text(angle=45,hjust=1,vjust=1)) +
  theme(legend.position="none")
#   coord_cartesian(ylim = c(-5, 5)) +
```

The matrix is scaled, so the enrichment profiles for each disease/phenotype have mean = 0 and SD = 1. 

```{r eval=TRUE}
# mtx.scaled <- mtx %>% t %>% scale %>% t # Row-wise (feature-wise) centering
mtx.scaled <- mtx %>% scale # Column-wise (sample-wise) centering
```

```{r}  
# Check how the scaled data looks like
ggplot(reshape2::melt(mtx.scaled), aes(x=Var2, y=value, fill=Var2)) + 
  geom_boxplot() + 
  theme(axis.text.x=element_text(angle=45,hjust=1,vjust=1)) +
  theme(legend.position="none")
#   coord_cartesian(ylim = c(-5, 5)) +
```

```{r}
# Multidimensional scaling
d <- mtx.scaled %>% t %>% dist # Euclidean distance on individual scaled, transposed matrix
#d <- as.dist(1 - cor(mtx.scaled, method = "spearman")) # Alternatively, correlation matrix
mds.coor <- cmdscale(d) # Classical MDS
mds.coor <- left_join(data.frame(Name=rownames(mds.coor), mds.coor), data.frame(Name=names(categoryNames), Category=categoryNames), by = c("Name" = "Name"))
ggplot(data=mds.coor, aes(x=X1, y=X2, label=Name)) +
  geom_point(aes(color = Category), size = 3) +
  geom_text(colour = "black", alpha = 0.8, size = 4, vjust = 2, hjust = 0, angle=0) +
  geom_hline(yintercept = 0, colour = "gray65") +
  geom_vline(xintercept = 0, colour = "gray65")
```

### Principal components analysis

Proportion of variance explained for the first 10 principle components, and a plot of the variance against the number of the principal components (screeplot).

```{r eval=TRUE, results='show'}
pca <- prcomp(t(mtx.scaled))
screeplot(pca, type="lines", main="PCA of the regulatory similarities")
summary(pca)$importance[, 1:10] %>% round(digits = 2) %>% kable
```

First two principal components

```{r eval=TRUE, fig.height=7, fig.width=11}
library(ggplot2)
scores <- as.data.frame(pca$x)
scores <- left_join(data.frame(Name=rownames(scores), scores), data.frame(Name=names(categoryNames), Category=categoryNames), by = c("Name" = "Name"))
ggplot(data=scores, aes(x=PC1, y=PC2, label=Name)) +
  ggtitle("PCA of the regulatory similarities") + 
  theme(plot.title = element_text(lineheight = 0.8, face="bold")) +
  scale_color_manual(values=c("red", "green", "blue", "black")) +
  geom_point(aes(color = Category), size = 4) +
  geom_hline(yintercept = 0, colour = "gray65") +
  geom_vline(xintercept = 0, colour = "gray65") +
  geom_text(colour = "black", size = 4, vjust = -1) +
  scale_x_continuous(limits = c(-60, 70))
#, vjust = 2, hjust = 0
```

```{r eval=FALSE, fig.height=6}
# First three principal components
library(scatterplot3d)
s3d <- scatterplot3d(scores[, c("PC1", "PC2", "PC3")],
              main="PCA of the regulatory similarities",
              angle=40,
              pch=19,
              cex.symbols = 1.2,
              color=as.numeric(factor(scores$Category)))
legend(x=-3, y=-1, legend=c("immunologic", "metabolic", "neurologic", "other"), col=c("red", "green", "blue", "black"), pch = 19)
```

```{r}
# Principal component analysis in 3D
# http://davetang.org/muse/2015/02/12/animated-plots-using-r/
rename <- function(x){
  if (x < 10) {
    return(name <- paste('000',i,'plot.png',sep=''))
  }
  if (x < 100 && i >= 10) {
    return(name <- paste('00',i,'plot.png', sep=''))
  }
  if (x >= 100) {
    return(name <- paste('0', i,'plot.png', sep=''))
  }
}
frames <- 360
#loop through plots
for(i in 1:frames){
  name <- rename(i)
  #saves the plot as a .png file in the working directory
  png(name)
  s3d <- scatterplot3d(scores[, c("PC1", "PC2", "PC3")],
                main=paste("Angle", i),
                angle=i,
                pch=19,
                cex.symbols=0.5,
                color=as.numeric(factor(scores$Category)))
  s3d.coords <- s3d$xyz.convert(scores[, c("PC1", "PC2", "PC3")])
  text(s3d.coords$x, s3d.coords$y, labels=scores$Name, pos=2, offset=0.5, cex=0.7)
  dev.off()
}
 
my_command <- 'convert *.png -delay 1 -loop 0 3d.gif'
system(my_command)
#system("rm *.png")
scatterplot3d(scores[, c("PC1", "PC2", "PC3")], color=as.numeric(factor(scores$Category)))
```

### Hierarchical clustering

Prepare the matrix of correlation coefficients, and cluster it.

```{r eval=TRUE}
## Using spearman correlations
# rcorr returns a list, [[1]] - correl coeffs, [[3]] - p-values. Type - pearson/spearman
mtx.cor <- Hmisc::rcorr(mtx.scaled, type=cortype)[[1]]
# Optionally, try Spearman or Kendall correlation
# mtx.cor[[1]] <- cor(as.matrix(mtx), method="kendall")
# Or, MIC
# library(minerva) # Playing with MIC correlation coefficient
# mtx.cor <- mine(mtx, n.cores=4)$MIC
# Or, using distance metric
# "euclidean", "maximum", "manhattan", "canberra", "binary" or "minkowski"
# dist.method <- "euclidean"
# mtx.cor <- dist(t(mtx), method = dist.method) # Get distance matrix

## Clustering
hclustergram <- hclust(as.dist(1-mtx.cor), method=hclust.method) # Cluster it
```

Get the minimum and maximum correlations among all diseases

```{r eval=TRUE, results='show'}
kable(head(gr_maxmin(mtx.scaled, cortype="spearman", "results/maxmin_corr_all.txt")))
```

```{r eval=TRUE, fig.height=8, fig.width=10}
# Plotting with aheatmap
suppressMessages(library(NMF))
suppressMessages(library(dendextend))
annot <- data.frame(Category=term.mapping$Category) #, Size=as.numeric(term.mapping$BEDcount))
annotColor <- list(Category=c("red", "green", "blue", "black")) #, Size=c("red", "white", "blue"))
hclustergram.ext <- hclustergram %>% as.dendrogram %>% set("branches_k_color", k = 2) %>% set("branches_lwd", 2) %>% ladderize
h <- aheatmap(mtx.cor, Rowv=hclustergram.ext, Colv=hclustergram.ext, color=col3(20), annCol=annot, annColors = annotColor, fontsize = 10, cexRow = 1, cexCol = 1)
```

```{r epigenomicVisualization1, fig.height=6, eval=FALSE}
# Quick visualization of the pre-clustered correlation matrix
h <- heatmap.2(mtx.cor, symkey = TRUE, Rowv=as.dendrogram(hclustergram), Colv=as.dendrogram(hclustergram), dendrogram="both", trace="none", density.info="none", col=color, RowSideColors=ColSideColors, ColSideColors=ColSideColors)  
```

```{r eval=FALSE}
# Example of testing combinations of clustering methods
dist.methods <- c("euclidean", "maximum","manhattan", "minkowski")
hclust.methods <- c("ward.D", "ward.D2", "single", "complete", "average", "mcquitty", "median", "centroid")
pdf("results/clustering_combinations.pdf")
for (d in dist.methods) {
  for (h in hclust.methods) {
    hclustergram <- hclust(as.dist(1-mtx.cor), method=h) # Cluster it
    par(oma=c(10,0,0,10), mar=c(10, 4.1, 4.1, 10), cex.main=0.65) # Adjust margins
    h<-heatmap.2(as.matrix(mtx.cor), Rowv=as.dendrogram(hclustergram), Colv=as.dendrogram(hclustergram), trace="none", density.info="none", col=color,  cexRow=1, cexCol=1,  main=paste(d, h),  RowSideColors=ColSideColors) 
    # distfun=function(x){dist(x, method=d)}, hclustfun=function(x){hclust(x, method=h)},
    # cellnote=formatC(as.matrix(mtx.cor), format="f", digits=2), notecol="black", notecex=1,
  }
}
dev.off()
```

We observe immunological diseases being grouped into a cluster distinct from the other diseases/traits. Within the cluster of immunological diseases, two or three subgroups can be noted. We focus our attention on this cluster and the subgroups of 21 immunological diseases to identify potential regulatory differences associated with the corresponding SNP sets.

```{r eval=TRUE}
ind <- categoryNames == "immunologic" # Indexes of immunological diseases
```

Get the minimum and maximum correlations among immunologic diseases

```{r eval=TRUE, results='show'}
kable(head(gr_maxmin(mtx.scaled[, ind], cortype="spearman", "results/maxmin_corr_imm.txt")))
```

```{r eval=TRUE, fig.height=7}
library(pvclust)
# # A function to calculate Spearman correlations
# spm <- function(x) {
#   res <- 1 - cor(x, method = "spearman", use = "pairwise.complete.obs")
#   res <- as.dist(res)
#   attr(res, "method") <- "spearman"
#   return(res)
# }
# library(parallel)
# cl <- makeCluster(4, type="PSOCK")
# set.seed(1)
# result <- parPvclust(cl=cl, mtx.scaled[, ind], method.dist=spm, method.hclust=hclust.method, nboot=10000)
# stopCluster(cl)
# saveRDS(result, "data/pvClust_03.Rds")
# # result <- readRDS("data/pvClust_01.Rds") # p-vals, ind-subset, method.dist="correlation", method.hclust=hclust.method, nboot=10000
# # result <- readRDS("data/pvClust_02.Rds") # p-vals, ind-subset, method.dist=spm, method.hclust=hclust.method, nboot=10000
result <- readRDS("data/pvClust_03.Rds") # z-scores, ind-subset, method.dist=spm, method.hclust=hclust.method, nboot=10000
dend <- as.dendrogram(result) %>% set("branches_k_color", k = 2) %>% set("branches_lwd", 2) %>% plot(main="Cluster dendrogram with AU/BP values (%)\n reproduced plot with dendrogram")
result %>% text
result %>% pvrect(alpha=0.89)
hclustergram.imm <- result$hclust # Make the heatmap plot whatever hclust object
```

```{r eval=TRUE, fig.height=6}
mtx.cor.imm <- mtx.cor[ind, ind]
# hclustergram.imm <- hclust(as.dist(1-mtx.cor.imm), method=hclust.method) #  Cluster it
hclustergram.imm.ext <- hclustergram.imm %>% as.dendrogram %>% set("branches_k_color", k = 2) %>% set("branches_lwd", 2) %>% ladderize
h <- aheatmap(mtx.cor.imm, Rowv=hclustergram.imm.ext, Colv=hclustergram.imm.ext, color=col3(20), fontsize = 10, cexRow = 1, cexCol = 1)
```

```{r eval=FALSE, fig.height=9}
# Another look at the correlation heatmap
library(corrplot)
par(oma=c(0,0,8,0), mar=c(5.1, 4.1, 8.1, 5.1)) # Adjust margins
corrplot(mtx.cor.imm, method = "color", type = "upper", col = col3(20), addCoef.col="black",
         order = "hclust", hclust.method = "ward", 
         tl.cex = 0.8, cl.pos = "r", diag = FALSE)
```

```{r eval=FALSE}
# Testing cluster number with COMMUNAL
library(COMMUNAL) 
ks <- seq(2,6)
result <- COMMUNAL(data=mtx.scaled[, ind], ks=ks, dist.metric = "correlation", aggl.method = "ward")
result.list <- list(list(result), ncol(mtx.scaled))
goodAlgs <- c("hierarchical", "kmeans")
goodMeasures <- c('wb.ratio', 'avg.silwidth', 'dunn')
values <- plotRange3D(result.list, ks, goodAlgs, goodMeasures)
# Testing cluster number with ConsensysClusterPlus
library(ConsensusClusterPlus)
results = ConsensusClusterPlus(mtx.scaled[, ind], maxK=6, reps=50, pItem=0.8, pFeature=1, title="ConsensusClusterPlus", clusterAlg="hc", distance="spearman", seed=1262118388.71279, plot="png")
# Store hclust object
hclustergram <- results[[3]][["consensusTree"]]
```

```{r fig.height=5, fig.width=10, eval=TRUE}
# Empirically define clusters
par(mfrow=c(1, 2))
par(oma=c(0,0,0,7), mar=c(5.1, 4.1, 4.1, 7.1)) # Adjust margins
h$Colv %>% as.dendrogram %>% color_labels(k=2) %>% set("labels_cex", 0.8) %>% set("branches_k_color", k = 2) %>% set("branches_lwd", 2)  %>% plot(horiz = TRUE) # %>% rect.dendrogram(k=3, border = 8, lty = 5, lwd = 2)
# plot(as.dendrogram(h$Colv, hang=-1), main = "Clustering dendrogram") # Plot dendrogram
# cl_num <- 2 # Empirically set desired numter of clusters
# cols <- rainbow(cl_num) # Make colors for clusters
# rect.hclust(as.hclust(h$Colv), k=cl_num, border=cols) # Define the clusters by rectangles
hcut <- 1.3 # Empirically set height to cut the tree
abline(v=hcut)
as.hclust(h$Colv)$height %>% density %>% plot(main = "Density of the clustered objects across the dendrogram height")
abline(v=hcut)
mtx.clust <- h$Colv %>% gr_clusters(height=hcut, minmembers=3, fileName="results/clustering_imm.txt")
par(mfrow=c(1, 1))
# Save the results of clustering
#write.table(as.data.frame(mtx.clust), "results/clustering_all.txt", sep="\t", row.names=FALSE, quote=FALSE)
```

## What are the Differentially Expressed Genomic Features (DEGFs) in each cluster

```{r eval=TRUE}
enrich <- function(postfix = "encBroadHMM", subset = "none") {
  mtx.degs <- gr_load_data(paste("data.gr.05/grweb_", postfix, "/matrix_PVAL.txt", sep=""), p2z = TRUE, subset)
  mtx.degs <- mtx.degs[, match(term.mapping$BED, colnames(mtx.degs))] # Match and subset the data
  colnames(mtx.degs) <- term.mapping$Name
  res <- gr_degfs(mtx.degs[, mtx.clust$eset.labels], mtx.clust, cutoff.pval = 0.1, cutoff.adjust = "fdr", p2z = TRUE, fileName = paste("results/degfs_", postfix, ".xlsx", sep=""))
  if (length(res) > 0) {
    num.rows <- ifelse(nrow(res[[1]]) < 10, nrow(res[[1]]), 10)
    kable(res[[1]][1:num.rows, c(1, 2, 3, 4, 10, 11, 12, 13)])
  } else {
    print("Nothing significant")
  }
}
```

### ENCODE Broad HMM

```{r eval=TRUE, results='show'}
enrich("encBroadHMM")
```

### ENCODE ChromCombined

```{r eval=TRUE, results='show'}
enrich("encChromCombined")
```

### ENCODE ChromHMM

```{r eval=TRUE, results='show'}
enrich("encChromHmm")
```

### ENCODE Segway

```{r eval=TRUE, results='show'}
enrich("encSegway")
```

### ENCODE DNAse

```{r eval=TRUE, results='show'}
enrich("encDNase")
```

### ENCODE Histone

```{r eval=TRUE, results='show'}
enrich("encHistone")
```

### ENCODE TFBS cellspecific

```{r eval=TRUE, results='show'}
enrich("encTFBS_cellspecific")
```

### ENCODE TFBS clustered

```{r eval=TRUE, results='show'}
enrich("encTFBS_clustered")
```

### chromStates15

```{r eval=TRUE, results='show'}
enrich(postfix = "roadmapChrom15", subset = cells_blood_all)
```

# Roadmap Epigenomics

### chromStates18

```{r eval=TRUE, results='show'}
enrich(postfix = "roadmapChrom18", subset = cells_blood_all)
```

### chromStates25

```{r eval=TRUE, results='show'}
enrich(postfix = "roadmapChrom25", subset = cells_blood_all)
```

### DNase_bPk-processed

```{r eval=TRUE, results='show'}
enrich(postfix = "roadmapDNase_bPk-processed", subset = cells_blood_all)
```

### DNase_nPk-processed

```{r eval=TRUE, results='show'}
enrich(postfix = "roadmapDNase_nPk-processed", subset = cells_blood_all)
```

### DNase_gPk-imputed

```{r eval=TRUE, results='show'}
enrich(postfix = "roadmapDNase_gPk-imputed", subset = cells_blood_all)
```

### DNase_nPk-imputed

```{r eval=TRUE, results='show'}
enrich(postfix = "roadmapDNase_nPk-imputed", subset = cells_blood_all)
```

### Histone_bPk-processed

```{r eval=TRUE, results='show'}
enrich(postfix = "roadmapHistone_bPk-processed", subset = cells_blood_all)
```

### Histone_gPk-processed

```{r eval=TRUE, results='show'}
enrich(postfix = "roadmapHistone_gPk-processed", subset = cells_blood_all)
```

### Histone_nPk-processed

```{r eval=TRUE, results='show'}
enrich(postfix = "roadmapHistone_nPk-processed", subset = cells_blood_all)
```

### Histone_gPk-imputed

```{r eval=TRUE, results='show'}
enrich(postfix = "roadmapHistone_gPk-imputed", subset = cells_blood_all)
```

### Histone_nPk-imputed

```{r eval=TRUE, results='show'}
enrich(postfix = "roadmapHistone_nPk-imputed", subset = cells_blood_all)
```



```{r eval=FALSE}
## Cell type-specific analysis
fname <- c("data.gr.05/grweb_roadmapDNase_gPk-imputed//matrix_PVAL.txt", "data.gr.05/grweb_roadmapDNase_nPk-imputed//matrix_PVAL.txt", "data.gr.05/grweb_roadmapHistone_gPk-imputed//matrix_PVAL.txt", "data.gr.05/grweb_roadmapHistone_nPk-imputed//matrix_PVAL.txt") # Imputed-ALL
mtx <- gr_load_data(fname, p2z = TRUE)
res <- gr_cellspecific(mtx, p2z = TRUE, fileName="results/cellspecific.xlsx")
```



```{r eval=FALSE}
# Testing random chance, Using GENES OR chromStates18+encHistone+encBroadHmm
# ---
mtx.degs <- load_gr_data(c("data.gr/2Roadmap_broadPeak/matrix_OR.txt", "data.gr/2Roadmap_DNase_hotspotbroadall/matrix_OR.txt", "data.gr/2ENCODE_Tfbs/matrix_OR.txt"))
mtx.degs <- mtx.degs[, match(term.mapping$BED, colnames(mtx))]
sum(mtx.degfs(mtx.degs[, mtx.clust$eset.labels], mtx.clust))
# Random simulations
mtx.rnd <- mtx.degs %>% mtx.rand(randomize = "row")
mtx.rnd <- mtx.degs %>% mtx.rand(randomize = "col")
mtx.rnd <- mtx.degs %>% mtx.rand(randomize = "mix")
mtx.rnd <- mtx.degs %>% mtx.rand(randomize = "rnd")
sum(mtx.degfs(mtx.rnd[, mtx.clust$eset.labels], mtx.clust))
```



