Genetic and epigenetic fine mapping of causal autoimmune disease variants
===

[1. Farh KK-H, Marson A, Zhu J, Kleinewietfeld M, Housley WJ, Beik S, Shoresh N, Whitton H, Ryan RJH, Shishkin A a., Hatan M, Carrasco-Alfonso MJ, Mayer D, Luckey CJ, Patsopoulos N a., De Jager PL, Kuchroo VK, Epstein CB, Daly MJ, Hafler D a., Bernstein BE: Genetic and epigenetic fine mapping of causal autoimmune disease variants. Nature 2014.](http://www.nature.com/nature/journal/vaop/ncurrent/pdf/nature13835.pdf)
---

[Supplemental table 1](http://www.nature.com/nature/journal/vaop/ncurrent/extref/nature13835-s1.xls) has genomic coordinates of disease-associated SNPs.

Prerequisites: Download [Supplemental table 1](http://www.nature.com/nature/journal/vaop/ncurrent/extref/nature13835-s1.xls) and save it as text file into a folder, such as *bed*.

Copy *Makefile* into that folder and run `make`. It will extract disease-specific SNPs into separate BED files.

*Makefile* also contains a command for making pair-wise overlap counts, stored in `overlapMatrix.txt` file.

[del Rosario RC-H, Poschmann J, Rouam SL, Png E, Khor CC, Hibberd ML, Prabhakar S: Sensitive detection of chromatin-altering polymorphisms reveals autoimmune disease mechanisms. Nat Methods 2015, 12.](http://www.nature.com/nmeth/journal/v12/n5/full/nmeth.3326.html)
---

*nmeth.3326-S2.xlsx* - Supplementary Table 3: List of haQTLs. The table contains the hg19 coordinates of the haQTLs, the raw P-values, the FDR Q-values and the parameters ascertained by the G-SCI test.

*nmeth.3326-S3.xlsx* - Supplementary Table 5: List of autoimmune GWAS SNPs. Sheet 1 (GWAS SNPs in LD with QTLs): autoimmune GWAS SNPs that are in strong LD with the regulatory QTL sets. Sheet 2 (All GWAS SNPs): the 831 autoimmune GWAS SNPS used in the analysis.